File: baseLib/src/test/java/me/ycdev/android/lib/common/type/LongHolderTest.java
package me.ycdev.android.lib.common.type;

import org.junit.Test;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

public class LongHolderTest {
    @Test
    public void basic() {
        {
            LongHolder holder = new LongHolder(0L);
            assertThat(holder.value, is(0L));
        }

        {
            LongHolder holder = new LongHolder(-10L);
            assertThat(holder.value, is(-10L));
        }

        {
            LongHolder holder = new LongHolder(100L);
            assertThat(holder.value, is(100L));
        }
    }
}

File: baseLib/src/test/java/me/ycdev/android/lib/common/type/LongHolderTest.kt
package me.ycdev.android.lib.common.type

import com.google.common.truth.Truth.assertThat
import org.junit.Test

class LongHolderTest {
    @Test
    fun basic() {
        run {
            val holder = LongHolder(0L)
            assertThat(holder.value).isEqualTo(0)
        }

        run {
            val holder = LongHolder(-10L)
            assertThat(holder.value).isEqualTo(-10L)
        }

        run {
            val holder = LongHolder(100L)
            assertThat(holder.value).isEqualTo(100L)
        }
    }
}
